src/client/components/filter-menu/time-filter-menu/time-filter-menu.tsx

```diff
@@ -43,31 +43,26 @@ export interface TimeFilterMenuProps {
   inside?: Element;
 }
 
+enum TimeFilterTab { RELATIVE = "relative", FIXED = "fixed"}
+
 export interface TimeFilterMenuState {
-  tab?: string;
+  tab: TimeFilterTab;
 }
 
-export class TimeFilterMenu extends React.Component<TimeFilterMenuProps, TimeFilterMenuState> {
-
-  private static readonly RELATIVE_TAB = "relative";
-  private static readonly FIXED_TAB = "fixed";
-
-  public mounted: boolean;
+function initialTab({ essence: { filter }, dimension: { name } }: TimeFilterMenuProps): TimeFilterTab {
```

Comment:
This name doesn't look like a function.

Reply:
function names: noun is getter, verb is side-effect

Reply:
😱